lib/stat_tracker.rb

```diff
@@ -80,13 +94,50 @@ def team_info
   end
 
 #---------------------------
-  private
+  # private
 
-  def coach_list_wins_losses
-    coach_hash = Hash.new
-    game_teams.each do |gt|
-      (coach_hash[gt.head_coach] ||= []) << gt.result
+def game_team_results_by_season(season)
+    games_of_season = games.find_all do |game|
+      game['season'] == season
+    end
+    game_ids_in_season = games_of_season.map do |game|
+      game['game_id']
+    end
+    @games_results_per_season = game_teams.find_all do |gt|
+      game_ids_in_season.include? gt['game_id']
     end
-    coach_hash
-  end  
+  end
+
+  def coaches_records_start
+    @coach_record_hash = {}
+    @games_results_per_season.each do |gr|
+      @coach_record_hash[gr['head_coach']] = {wins: 0, losses: 0, ties:0}
+    end
+    @coach_record_hash
+  end
```

Comment:
Good job on finally getting this one @Stearnzy! This one definitely looked tough...

The one question I have would be about the use of instance variables and your thought process behind that. If you needed `coach_record_hash`, for example, then you could just call the coaches records_start method